### PR TITLE
Reorganize math headers

### PR DESCRIPTION
**Pull request by bernhardmgruber**
This PR merges all math headers into 3 headers. One for the standard library functions, one for CUDA/HIB functions and one for the traits. This merge improves future refactorings a lot as editing becomes more local.

### CODE REVIEW

**Comment by bernhardmgruber** [comment]
The sphinx build chokes on the `é` in `René` ...

**Comment by psychocoderHPC** [comment]
> The sphinx build chokes on the `é` in `René` ...

Feel free to change it to `Rene` normally I use only ASCII to avoid such issues.

**Comment by bernhardmgruber** [comment]
> > The sphinx build chokes on the `é` in `René` ...
> 
> Feel free to change it to `Rene` normally I use only ASCII to avoid such issues.

Thx! The docs build are passing now!

**Comment by j-stephan** [comment]
Which problem is solved by this?

**Comment by bernhardmgruber** [comment]
> Which problem is solved by this?

Maintainablity via refactorings.
Example: I wanted to collapse the namespaces:
```c++
namespace alpaka {
    namespace math {
    ...
    }
}
```
today into:
```c++
namespace alpaka::math {
    ...
}
```
I started editing 52 (!!) files until I realized, that everybody has to pay that cost whenever we modify something across all math functions.
Further example: @fwyzard added:
```c++
#    if !defined(ALPAKA_HOST_ONLY)

#        if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) && !BOOST_LANG_CUDA
#            error If ALPAKA_ACC_GPU_CUDA_ENABLED is set, the compiler has to support CUDA!
#        endif

#        if defined(ALPAKA_ACC_GPU_HIP_ENABLED) && !BOOST_LANG_HIP
#            error If ALPAKA_ACC_GPU_HIP_ENABLED is set, the compiler has to support HIP!
#        endif
....
#endif
```
to all CUDA math functions, which are 27 places in 27 files. With this PR, it would have been 1 place in 1 file.

File: include/alpaka/math/MathStdLib.hpp [hunk 1]
```diff
@@ -1,4 +1,5 @@
-/* Copyright 2019 Benjamin Worpitz
+/* Copyright 2022 Alexander Matthes,Axel Huebl, Benjamin Worpitz, Matthias Werner, Bernhard Manfred Gruber, Jeffrey
+ * Kelling
```
Comment (j-stephan) [inline include/alpaka/math/MathStdLib.hpp]:
```suggestion
/* Copyright 2022 Alexander Matthes,Axel Huebl, Benjamin Worpitz, Matthias Werner, Bernhard Manfred Gruber, 
 * Jeffrey Kelling
```

File: include/alpaka/math/MathUniformCudaHipBuiltIn.hpp [hunk 2]
```diff
@@ -1,4 +1,5 @@
-/* Copyright 2019 Benjamin Worpitz
+/* Copyright 2022 Axel Huebl, Benjamin Worpitz, Matthias Werner, Bert Wesarg, Valentin Gehrke, Rene Widera, Jan
+ * Stephan, Andrea Bocci, Bernhard Manfred Gruber, Jeffrey Kelling
```
Comment (j-stephan) [inline include/alpaka/math/MathUniformCudaHipBuiltIn.hpp]:
```suggestion
/* Copyright 2022 Axel Huebl, Benjamin Worpitz, Matthias Werner, Bert Wesarg, Valentin Gehrke, Rene Widera,
 * Jan Stephan, Andrea Bocci, Bernhard Manfred Gruber, Jeffrey Kelling
```

File: test/unit/math/src/sincos.cpp [hunk 3]
```diff
@@ -7,7 +7,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
```
Comment (j-stephan) [inline test/unit/math/src/sincos.cpp:7]: The copyright header needs an update here.

**Review by j-stephan — CHANGES_REQUESTED** [review]
I'm not convinced that I want this but I'll follow the majority here. Anyway, a few things that need to be changed in any case:

File: include/alpaka/math/MathStdLib.hpp (hunk 1 above)
Comment (bernhardmgruber) [reply include/alpaka/math/MathStdLib.hpp]: fixed

File: include/alpaka/math/MathUniformCudaHipBuiltIn.hpp (hunk 2 above)
Comment (bernhardmgruber) [reply include/alpaka/math/MathUniformCudaHipBuiltIn.hpp]: fixed

File: test/unit/math/src/sincos.cpp (hunk 3 above)
Comment (bernhardmgruber) [reply test/unit/math/src/sincos.cpp:7]: done

File: include/alpaka/math/MathStdLib.hpp [hunk 4]
```diff
@@ -1,4 +1,5 @@
-/* Copyright 2019 Benjamin Worpitz
+/* Copyright 2022 Alexander Matthes,Axel Huebl, Benjamin Worpitz, Matthias Werner, Bernhard Manfred Gruber,
```
Comment (j-stephan) [inline include/alpaka/math/MathStdLib.hpp]:
```suggestion
/* Copyright 2022 Alexander Matthes, Axel Huebl, Benjamin Worpitz, Matthias Werner, Bernhard Manfred Gruber,
```

**Review by j-stephan — CHANGES_REQUESTED** [review]
4321 lines of code removed. I do enjoy that. One small whitespace issue and we're good to go:

**Comment by bernhardmgruber** [comment]
> > The sphinx build chokes on the `é` in `René` ...
> 
> Feel free to change it to `Rene` normally I use only ASCII to avoid such issues.

It turns out the failure was caused by me since I preprared this PR on a Windows machine and the encoding might have been wrong. I switched back to René on a Linux machine, with utf-8 encoding, and the docs are building fine.

File: include/alpaka/math/MathStdLib.hpp (hunk 4 above)
Comment (bernhardmgruber) [reply include/alpaka/math/MathStdLib.hpp]: fixed

**Review by j-stephan — APPROVED** [review]

**Pull request merged**